spark_pipeline_framework/transformers/connecthub_data_receiver/v1/connecthub_data_receiver.py

```diff
@@ -0,0 +1,171 @@
+            int
+        ] = 0,  # setting to 0 ignores limit and will return all results
+        name: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        progress_logger: Optional[ProgressLogger] = None,
+    ) -> None:
+        """
+        Transformer to call and receive data from ConnectHub (DocumentDB). It is specific to ConnectHub for now,
+        but it should/will get refactored into a generic DocumentDbDataReceiver when time allows.
+
+        :param view_name: name of the view to read the response into
+        :param conn_string: the root URL
+        :param last_run_date: (Optional) the datetime which we should be querying from to get the latest updates. It is
+        set to 1/1/1970 by default so that we grab all records without filtering (since there doesn't seem to be a good
+        automated way to pass this in from pipeline runs at this point)
+        :param page_size: (Optional) the number of documents we should be querying at a time, if we want paging.
+        Setting to 0 (which is the default) will ignore paging and all documents (within the query parameters) will
+        get pulled at once.
+        :param name: (Optional) name of transformer
+        :param parameters: (Optional) parameters
+        :param progress_logger: (Optional) progress logger
+        """
+        super(ConnectHubDataReceiver, self).__init__(
+            name=name,
+            parameters=parameters,
+            progress_logger=progress_logger,
+        )
+
+        self.logger = get_logger(__name__)
+
+        self.view_name: Param[str] = Param(self, "view_name", "")
+        self._setDefault(view_name=view_name)
+
+        self.page_size: Param[int] = Param(self, "page_size", "")
+        self._setDefault(page_size=page_size)
+
+        self.conn_string: Param[str] = Param(self, "conn_string", "")
+        self._setDefault(conn_string=conn_string)
+
+        self.last_run_date: Param[datetime] = Param(self, "last_run_date", "")
+        self._setDefault(last_run_date=last_run_date)
+
+    def _transform(self, df: DataFrame) -> DataFrame:
+        view_name = self.get_view_name()
+        conn_string = self.get_conn_string()
+
+        try:
+            client = pymongo.MongoClient(conn_string)  # type: ignore
+
+            integration_hub_db = client.integration_hub
+            client_connection = integration_hub_db.client_connection
+
+            converted_data: List[Any] = []
+            last_seen = None
+            while True:
+                data, last_seen = self.pagination(client_connection, last_seen)
+                if not data:
+                    break
+                converted_data.extend(data)
+
+            df2 = df.sql_ctx.read.json(
+                sc(df).parallelize([json.dumps(r) for r in converted_data])
+            )
+            df2.createOrReplaceTempView(view_name)
+
+            client.close()
+        except Exception as e:
+            self.logger.error(e)
```

Comment:
wondering if it makes sense to just swallow the exception here. if there is an error reading the data then the rest of the pipeline cannot continue. perhaps we should be throwing an exception. thoughts?

Reply:
Yes, and since I don't see any useful information I can add at this point, I think there's no reason for me to catch it in the first place. Just do try/finally to close the connection properly.